### ISSUE TITLE
The remote volumes will not be found when adding new nodes into minio distributed mode

### ISSUE THREAD

**Issue by afterhill** [minio/minio#4140]
When adding new new node into minio distributed modes,  the following error was repeated in living container and the entire minio system crashed:

```
$ sudo docker logs -f 4cfe
Created minio configuration file successfully at /root/.minio
time="2017-04-17T06:37:43Z" level=error msg="Failed to lookup host" cause="lookup minio1 on 127.0.0.11:53: server misbehaving" source="[object-api-common.go:149:isLocalStorage()]" 
.... repeat 20+ times....
Initializing data volume for first time. Waiting for other servers to come online (elapsed 22s)
.... repeat endlessly...
```

## Expected Behavior
The new node was merged properly.

## Current Behavior
The entire minio system will crash and all of running container will repeat the following error message:

```
$ sudo docker logs -f 4cfe
Initializing data volume for first time. Waiting for other servers to come online (elapsed 22s)
.... repeat endlessly...
```

## Possible Solution
Remove all of `.minio.sys` folder will bring the minio system up immediately!

## Steps to Reproduce (for bugs)
1. 2 virtual box VMs, 3 disks for each VM.
2.  $docker-machine ls:
    ```
    minio-swarm-01   -        generic   Running   tcp://10.25.3.11:2376           v17.04.0-ce   
    minio-swarm-02   -        generic   Running   tcp://10.25.3.12:2376           v17.04.0-ce 
    ```
3. lsblk VM1:

**/data1 has some existing data and /data1 has the directory .minio.sys, /data2 and /data3 are empty**

```
NAME   MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
sda      8:0    0    40G  0 disk 
└─sda1   8:1    0    40G  0 part /
sdb      8:16   0   1.9T  0 disk 
└─sdb1   8:17   0   1.9T  0 part /data1
sdc      8:32   0   1.9T  0 disk 
└─sdc1   8:33   0   1.9T  0 part /data2
sdd      8:48   0   1.9T  0 disk 
└─sdd1   8:49   0   1.9T  0 part /data3
```

4. lsblk VM2:

**All of /data1 /data2 /data3 are new and empty**

```
NAME   MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
sda      8:0    0    40G  0 disk 
└─sda1   8:1    0    40G  0 part /
sdb      8:16   0   1.9T  0 disk 
└─sdb1   8:17   0   1.9T  0 part /data1
sdc      8:32   0   1.9T  0 disk 
└─sdc1   8:33   0   1.9T  0 part /data2
sdd      8:48   0   1.9T  0 disk 
└─sdd1   8:49   0   1.9T  0 part /data3
```
5. compose file:
```
$ cat cluster-docker-compose.yml 
version: "3"

services:
  minio1:
    image: minio/minio
    container_name: minio_node01
    ports:
      - "9000"
    networks:
      - minio_distributed
    volumes:
      - /data1:/export1
      - /data2:/export2
      - /data3:/export3
    environment:
      MINIO_ACCESS_KEY: <secret>
      MINIO_SECRET_KEY: <secret>
    command: server http://minio1/export1 http://minio1/export2 http://minio1/export3 http://minio2/export1 http://minio2/export2 http://minio2/export3
    deploy:
      restart_policy:
        condition: on-failure
      placement:
        constraints:
          - node.hostname == minio-node-01

  minio2:
    image: minio/minio
    container_name: minio_node02
    ports:
      - "9000"
    networks:
      - minio_distributed
    volumes:
      - /data1:/export1
      - /data2:/export2
      - /data3:/export3
    environment:
      MINIO_ACCESS_KEY: <secret>
      MINIO_SECRET_KEY: <secret>
    command: server http://minio1/export1 http://minio1/export2 http://minio1/export3 http://minio2/export1 http://minio2/export2 http://minio2/export3
    deploy:
      restart_policy:
        condition: on-failure
      placement:
        constraints:
          - node.hostname == minio-node-02

networks:
  minio_distributed:
    driver: overlay
```
6. deploy via docker swarm
```
sudo docker $(docker-machine config minio-swarm-01) stack deploy --compose-file=cluster-docker-compose.yml cloud_store
```
7. All of nodes are swarmed:
```
$ sudo docker node ls
ID                           HOSTNAME            STATUS  AVAILABILITY  MANAGER STATUS
fnspxhumoydlqpkced36syzvk *  minio-swarm01-3-11  Ready   Active        Leader
mm0l8fs01cdfgrtbv3l1bv9tq    minio-swarm02-3-12  Ready   Active        Reachable
```

## Context
1. `/data1 /data2 /data3` under VM1 are have old folder `.minio.sys`

```
$ ls -al /data*
/data1:
total 12
drwxr-xr-x  3 root root 4096 Apr 17 10:34 .
drwxr-xr-x 25 root root 4096 Apr 17 03:16 ..
drwxr-xr-x  4 root root 4096 Apr 17 10:34 .minio.sys

/data2:
total 12
drwxr-xr-x  3 root root 4096 Apr 17 10:34 .
drwxr-xr-x 25 root root 4096 Apr 17 03:16 ..

/data3:
total 12
drwxr-xr-x  3 root root 4096 Apr 17 10:34 .
drwxr-xr-x 25 root root 4096 Apr 17 03:16 ..
```

2. `/data1 /data2 /data3` under VM2 are clean

```
$ ls -al /data*
/data1:
total 12
drwxr-xr-x  3 root root 4096 Apr 17 10:34 .
drwxr-xr-x 25 root root 4096 Apr 17 03:16 ..

/data2:
total 12
drwxr-xr-x  3 root root 4096 Apr 17 10:34 .
drwxr-xr-x 25 root root 4096 Apr 17 03:16 ..

/data3:
total 12
drwxr-xr-x  3 root root 4096 Apr 17 10:34 .
drwxr-xr-x 25 root root 4096 Apr 17 03:16 ..
```

## Your Environment
* docker ce stable latest
* minio stable latest

**Comment by harshavardhana** [comment]
Distributed setup wouldn't work with existing data on the previous volumes i.e in your case VM1 /data1 - we need to fix this to provide the message properly.  The situation you ended up with can clearly happen and should be easy to reproduce. 

So it is expected that you start with fresh volumes, as you stated once cleared it works properly. We will fix the messaging problem here.

**Comment by harshavardhana** [comment]
This issue is reproducible quite simply like this, choose a directory say `/tmp/1` run fs backend on it once. 

```
minio server /tmp/1
```

Kill the server and now attempt to use `/tmp/1` as part of the distributed set using the following script.

```
#!/bin/bash

$(minio server --address localhost:9001 http://localhost:9001/tmp/1 http://localhost:9002/tmp/2 http://localhost:9003/tmp/3 http://localhost:9004/tmp/4)&
$(minio server --address localhost:9002 http://localhost:9001/tmp/1 http://localhost:9002/tmp/2 http://localhost:9003/tmp/3 http://localhost:9004/tmp/4)&
$(minio server --address localhost:9003 http://localhost:9001/tmp/1 http://localhost:9002/tmp/2 http://localhost:9003/tmp/3 http://localhost:9004/tmp/4)&
minio server --address localhost:9004 http://localhost:9001/tmp/1 http://localhost:9002/tmp/2 http://localhost:9003/tmp/3 http://localhost:9004/tmp/4
```

One of the last servers failed to start where as the other servers are waiting on the connection for quorum, which is exactly what happened in this scenario. 
```
$ ~/run-dist.sh 
FATA[0000] Initializing object layer failed              cause=Unsupported backend format [fs] found source=[server-main.go:276:serverMain()]
FATA[0000] Initializing object layer failed              cause=Unsupported backend format [fs] found source=[server-main.go:276:serverMain()]
Initializing data volume. Waiting for minimum 3 servers to come online. (elapsed 20s)
Initializing data volume. Waiting for minimum 3 servers to come online. (elapsed 38s)
```

Now if any server that reaches this stage waits by design to have minimum of quorum number of servers to come online, say if other servers died.

**Comment by harshavardhana** [comment]
A fix has been merged to give enough time for the neighboring servers to come online and detect each others failure.

**Comment by harshavardhana** [comment]
Fixed by #4142
